src/main/java/eggy/ui/Ui.java

```diff
@@ -0,0 +1,133 @@
+package eggy.ui;
+
+import eggy.task.Task;
+import eggy.task.TaskList;
+
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * The user interface of the chatbot.
+ */
+public class Ui {
+    /** The divider line. */
+    private final String DIVIDER = "    ____________________________________________________________";
+    /** The scanner to read user input. */
+    private final Scanner sc = new Scanner(System.in);
+
+    /**
+     * Reads the next line of user input.
+     *
+     * @return The next line of user input.
+     */
+    public String readCommand() {
+        return sc.nextLine();
+    }
+
+    /**
+     * Prints the divider line.
+     */
+    public void printDivider() {
+        System.out.println(DIVIDER);
+    }
+
+    /**
+     * Prints the welcome message.
+     *
+     * @param name The name of the chatbot.
+     */
+    public void printWelcome(String name) {
+        this.printDivider();
+        System.out.println("     Hello! I'm " + name + "\uD83E\uDD5A.");
+        System.out.println("     What can I do for you?");
+        this.printDivider();
+    }
+
+    /**
+     * Prints the goodbye message.
+     */
+    public void printGoodbye() {
+        System.out.println("     Bye\uD83D\uDC4B. Hope to see you again soon!");
+        sc.close();
+    }
+
+    /**
+     * Prints the task added message.
+     *
+     * @param task The task added.
+     * @param taskCount The number of tasks in the task list.
+     */
+    public void printTaskAdded(Task task, int taskCount) {
+        System.out.println("     Got it. I've added this task:");
+        System.out.println("       " + task.toString());
+        System.out.println("     Now you have " + taskCount + " tasks in the list.");
+    }
+
+    /**
+     * Prints the task removed message.
+     *
+     * @param task The task removed.
+     * @param taskCount The number of tasks in the task list.
+     */
+    public void printTaskRemoved(Task task, int taskCount) {
+        System.out.println("     Noted. I've removed this task:");
+        System.out.println("       " + task.toString());
+        System.out.println("     Now you have " + taskCount + " tasks in the list.");
+    }
```

Comment:
I like the naming of methods in this class, their objective is clear from the names alone without having to look at the code in detail.